lib/intent-models/server-security.js

```diff
@@ -0,0 +1,23 @@
+const IntentBase = require('./Base');
+const IntentSecurityScheme = require('./security-scheme')
+
+/**
+ * 
+ * @class
+ * @alias module:@asyncapi/parser#IntentServerSecurity
+ * @extends IntentBase
+ * @returns {SecurityScheme}
+ */
+ class IntentServerSecurity extends IntentBase {
+    /**
+     * @returns {IntentSecurityScheme}
+     */
+    securityScheme() { return new IntentSecurityScheme();}
+    
+    /**
+     * @returns {string[]}
+     */
+    values() { return []; }
```

Comment:
Why should `values` not be part of `IntentSecurityScheme` and then we can remove `IntentServerSecurity` 🤔 

cc @smoya

Reply:
Because they represent different things. Please correct me if I'm wrong with the following:

A SecurityScheme can be reused by other servers. On the other hand, ServerSecurity holds the config values for a specific SecuritySchema that apply to a particular server.

Reply:
The way I see it is that with the spec is that you define server with a security schema, you define it as:

Server security specified as:
```
security: [
  {
     "api_key": []
  }
]
```

With a security component named `api_key`: 

```
{
  "type": "httpApiKey",
  "name": "api_key",
  "in": "header"
}
```

My intent is to get the security object (located in the security components). Where the so-called values (`[]`) are only used if the security scheme is of type "oauth2" or "openIdConnect", then the value is a list of scope names. 

So to me, it would make sense for `IntentSecurityScheme` to contain a scope names intent which is the previous `values` variable, and then remove `IntentServerSecurity` completely.

This generally makes more sense conceptually to me in terms of a general understanding of what the security model should contain. This current approach is more tuned towards the spec directly.

What do you think?

Reply:
Changed the mocks.